azure/scope/identity_test.go

```diff
@@ -135,241 +133,234 @@ func TestAllowedNamespaces(t *testing.T) {
+	certPEM, err := base64.StdEncoding.DecodeString(encodedCertData)
+	g.Expect(err).NotTo(HaveOccurred())
+
 	tests := []struct {
-		name     string
-		identity *infrav1.AzureClusterIdentity
-		want     bool
+		name                         string
+		cluster                      *infrav1.AzureCluster
+		secret                       *corev1.Secret
+		identity                     *infrav1.AzureClusterIdentity
+		ActiveDirectoryAuthorityHost string
 	}{
 		{
-			name: "user assigned identity",
+			name: "workload identity",
+			cluster: &infrav1.AzureCluster{
+				Spec: infrav1.AzureClusterSpec{
+					AzureClusterClassSpec: infrav1.AzureClusterClassSpec{
+						IdentityRef: &corev1.ObjectReference{
+							Kind: infrav1.AzureClusterIdentityKind,
+						},
+					},
+				},
+			},
 			identity: &infrav1.AzureClusterIdentity{
 				Spec: infrav1.AzureClusterIdentitySpec{
-					Type:       infrav1.UserAssignedMSI,
-					ResourceID: "my-resource-id",
+					Type:     infrav1.WorkloadIdentity,
+					ClientID: fakeClientID,
+					TenantID: fakeTenantID,
+					ClientSecret: corev1.SecretReference{
+						Name: "test-identity-secret",
+					},
+				},
+			},
+			secret: &corev1.Secret{
```

Comment:
Should we omit this secret entirely since we don't expect one to exist for Workload ID?